llvm/lib/Target/X86/X86PfmCounters.td

```diff
@@ -204,6 +204,27 @@ def : PfmCountersBinding<"icelake-server", IceLakePfmCounters>;
 def : PfmCountersBinding<"rocketlake", IceLakePfmCounters>;
 def : PfmCountersBinding<"tigerlake", IceLakePfmCounters>;
 
+def AlderLakePfmCounters : ProcPfmCounters {
+  let CycleCounter = UnhaltedCoreCyclesPfmCounter;
+  let UopsCounter = UopsIssuedPfmCounter;
+  let IssueCounters = [
+    PfmIssueCounter<"ADLPPort00", "uops_dispatched_port:port_0">,
+    PfmIssueCounter<"ADLPPort01", "uops_dispatched_port:port_1">,
+    PfmIssueCounter<"ADLPPort02", "uops_dispatched_port:port_2_3_10">,
+    PfmIssueCounter<"ADLPPort03", "uops_dispatched_port:port_2_3_10">,
+    PfmIssueCounter<"ADLPPort04", "uops_dispatched_port:port_4_9">,
```

Comment:
Use ADLPPort04_09 instead - adding the other ProcResGroup combos to X86SchedAlderlakeP.td shouldn't be a problem.

Reply:
Can't believe I missed that one along with 1-2 others. I've added in the ones that were missing as well to `X86SchedAlderlakeP.td`, along with a `ADLPPortAny` `ProcResGroup` so that all of the overlapping groups have a super group (I believe that won't cause any issues?).